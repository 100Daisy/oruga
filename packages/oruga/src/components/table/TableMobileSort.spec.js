import { shallowMount } from '@vue/test-utils'
import OTableMobileSort from '@components/table/TableMobileSort'

let wrapper

describe('OTableMobileSort', () => {
    beforeEach(() => {
        wrapper = shallowMount(OTableMobileSort, {sync: false})
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('OTableMobileSort')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    describe('showPlaceholder', () => {
        it('is true if there are no columns', () => {
            expect(wrapper.vm.showPlaceholder).toBeTruthy()
        })
    })

    it('set mobileSort value when currentSortColumn is changed', (done) => {
        wrapper.setProps({currentSortColumn: 'val'})
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.mobileSort).toEqual(wrapper.vm.currentSortColumn)
            done()
        })
    })

    it('emit sort event with mobileSort value when sort is called', () => {
        wrapper.vm.mobileSort = 'val'
        wrapper.vm.sort()
        const valueEmitted = wrapper.emitted()['sort'][0]
        expect(valueEmitted).toContainEqual(wrapper.vm.mobileSort)
    })
})
