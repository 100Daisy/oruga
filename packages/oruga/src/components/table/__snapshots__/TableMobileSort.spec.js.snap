// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTableMobileSort render correctly 1`] = `
<div class="o-table-mobile-sort">
    <o-field-stub addons="true" rootclass="o-field" horizontalclass="o-field-horizontal" labelclass="o-field-label" labelhorizontalclass="o-field-horizontal-label" contenthorizontalclass="o-field-horizontal-content" messageclass="o-field-message">
        <o-select-stub expanded="true" usehtml5validation="true" iconright="caret-down" rootclass="o-control-select" selectclass="o-select" roundedclass="o-select-rounded" multipleclass="o-select-multiple" emptyclass="o-select-empty">
            <!---->
        </o-select-stub>
        <o-button-stub variant="primary" nativetype="button" tag="button" rootclass="o-button" outlinedclass="o-button-outlined" invertedclass="o-button-inverted" expandedclass="o-button-expanded" roundedclass="o-button-rounded">
            <o-icon-stub icon="arrow-up" size="small" rootclass="o-icon" clickableclass="o-icon-clickable" spinclass="o-icon-spin" both="true" class="o-icon-sort-desc"></o-icon-stub>
        </o-button-stub>
    </o-field-stub>
</div>
`;
